src/status_im/chat/events/sign_up.cljs

```diff
@@ -49,37 +49,85 @@
+   (sms-listener/remove-sms-listener subscription)))
+
+(defn- sms-receive-handler [{confirmation-code :body}]
+  (when-let [matches (re-matches #"(\d{4})" confirmation-code)]
+    (re-frame/dispatch [::sign-up-confirm (second matches)])))
+
+(def start-listening-confirmation-code-sms-event
+  [:request-permissions
+   [:receive-sms]
+   (fn []
+     (let [listener (sms-listener/add-sms-listener sms-receive-handler)]
+       (re-frame/dispatch [:start-listening-confirmation-code-sms listener])))])
+
+(handlers/register-handler-fx
+ ::sign-up-success
+ [re-frame/trim-v (re-frame/inject-cofx :random-id)]
+ (fn [{:keys [db random-id]} message-id]
+   (-> {:db         db
+        :dispatch-n [;; create manual way for entering confirmation code
+                     (sign-up/enter-confirmation-code-event random-id)
+                     ;; create automatic way for receiving confirmation code
+                     start-listening-confirmation-code-sms-event]}
+       (message-seen message-id))))
+
+(handlers/register-handler-fx
+ :update-phone-number-in-profile
+ [re-frame/trim-v]
+ (fn [{{:keys [chats] :as db} :db}]
+   (let [phone-command (->> (get-in chats ["console" :messages])
+                            (filter (fn [{:keys [type content]}]
+                                      (and (= type :response)
+                                           (= (:command content) "phone"))))
+                            (first))]
+     {:db         db
+      :dispatch-n [[:account-update {:phone (get-in phone-command [:content :params :phone])}]]})))
+
+(defn- extract-last-phone-number [chats]
+  (let [phone-message (->> (get-in chats ["console" :messages])
+                           (filter (fn [{:keys [type content]}]
+                                     (and (= type :response)
+                                          (= (:command content) "phone"))))
+                           (first))]
+    (get-in phone-message [:content :params :phone])))
 
 (handlers/register-handler-fx
   ::sign-up-confirm
   (fn [{:keys [db]} [confirmation-code message-id]]
     (sign-up-confirm db confirmation-code message-id)))
 
 (defn- sign-up-confirmed [{:keys [db] :as fx}]
-  (cond-> (update fx :dispatch-n conj [:request-permissions
-                                       [:read-contacts]
-                                       #(re-frame/dispatch [:sync-contacts (fn [contacts]
-                                                                             [::contacts-synced contacts])])])
-    (:confirmation-code-sms-listener db)
-    (merge {:db                  (dissoc db :confirmation-code-sms-listener)
-            :remove-sms-listener (:confirmation-code-sms-listener db)})))
+  (let [last-phone-number (extract-last-phone-number (:chats db))]
+    (cond->
+      fx
+
+      last-phone-number
+      (update :dispatch-n conj [:account-update {:phone last-phone-number}])
+
+      true
+      (update :dispatch-n concat [[:stop-listening-confirmation-code-sms]
```

Comment:
The same as above, why instead of dispatching `:stop-listening-confirmation-code-sms` we just don't compose it as a function ?